colossus/src/main/scala/colossus/protocols/http/Header.scala

```diff
@@ -176,13 +176,51 @@ object Connection {
   }
 }
 
+case class QueryParameters(parameters: Seq[(String, String)]) extends AnyVal{
+
+  def apply(key: String) = singleValue(key).get
+
+  /**
+   * Get the value of a query string parameter when only at most one value is expected
+   **/
+  def singleValue(key: String): Option[String] = parameters.collectFirst{case (k,v) if k == key => v}
```

Comment:
how do you feel about getSingle and get (or maybe get and getAll since pulling a single value from a parameter is probably the default case)